src/CSET/operators/constraints.py

```diff
@@ -65,153 +64,45 @@ def generate_var_constraint(varname: str, **kwargs) -> iris.Constraint:
-    to constrain to specific model levels.
-
-    Arguments
-    ---------
-    model_level: int|list
-        CF compliant model levels.
-
-    Returns
-    -------
-    model_level_constraint: iris.Constraint
-    """
-    # Ensure is iterable.
-    if not isinstance(model_level, Iterable):
-        model_level = [model_level]
-
-    if len(model_level) == 0:
-        # If no levels specified reject cubes with any model level coordinate.
-        def no_model_level_number(cube):
-            # Check for coordinate existence.
-            return bool(cube.coords("model_level_number"))
-
-        return iris.Constraint(cube_func=no_model_level_number)
-
-    # LFRic data do not have a model_level_number coordinate like um data,
-    # but a full_levels or half_levels coordinate instead.
-    # Test if vertical dimension of cube contains full_levels
-    # coordinate or model_level_number coordinate to return correct
-    # vertical level constraint.
-    return iris.Constraint(model_level_number=model_level)
-
-
-def generate_half_level_constraint(
-    model_level: int | list[int], **kwargs
-) -> iris.Constraint:
-    """Generate constraint for particular half levels.
-
-    Operator that takes a CF compliant model_level_number int or list of int, and uses iris to
-    generate a constraint to be passed into the read operator to minimize the
-    CubeList the read operator loads and speed up loading.
-
-    Arguments
-    ---------
-    model_level: int | list[int]
-        CF compliant half levels.
-
-    Returns
-    -------
-    half_level_constraint: iris.Constraint
-    """
-    # Ensure is iterable.
-    if not isinstance(model_level, Iterable):
-        model_level = [model_level]
-
-    if len(model_level) == 0:
-        # If no levels specified reject cubes with any model level coordinate.
-        def no_half_levels(cube):
-            # Check for coordinate existence.
-            return bool(cube.coords("half_levels"))
-
-        return iris.Constraint(cube_func=no_half_levels)
-
-    # LFRic data do not have a model_level_number coordinate like um data,
-    # but a full_levels or half_levels coordinate instead.
-    # Test if vertical dimension of cube contains full_levels
-    # coordinate or model_level_number coordinate to return correct
-    # vertical level constraint.
-    return iris.Constraint(half_levels=model_level)
+    Operator that generates a constraint to constrain to specific model or pressure levels.
+    If no levels are specified then any cube with the specified coordinate is
+    rejected.
 
-
-def generate_full_level_constraint(
-    model_level: int | list[int], **kwargs
-) -> iris.Constraint:
-    """Generate constraint for particular full levels.
-
-    Operator that generates a constraint to be passed into the read operator
-    to constrain to specific full levels.
+    Typically ``coordinate`` will be ``"pressure"`` or ``"model_level_number"``
```

Comment:
are there cases where we also have "model_level"?

Reply:
I've not seen any files with that, but you can give it as the coordinate for this operator.